subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsActionFactory.kt

```diff
@@ -50,4 +50,8 @@ interface InstrumentationTestsActionFactory {
 
         override fun provideFinalizer() = finalizerFactory.create()
     }
+
+    companion object {
+        val gson by lazy { GsonBuilder().setPrettyPrinting().create() }
```

Comment:
it looks like this gson producer used everywhere in plugin, so should be passed as arguments or placed globally, not in concrete `InstrumentationTestsActionFactory` companion

Reply:
I agree, it's common question: How to share common components like the `gson` in the plugin.
The ideal world is when we set everything via object constructors but we live with Gradle and to do this I have to write much code.
Dependency Inversion solves two problems: reusing and simple changing dependencies for tests for example.

In that case, I don't see any reason for refactoring that code because I don't need to change implementations now.
And I see `InstrumentationTestsActionFactory` as a normal place for `gson` constant which used through the plugin.

Could you tell me how you image `gson` at the package level?

Reply:
Just some singleton(object) for global val's for instrumentation plugin